### PR TITLE
Colossus cleanup fix

### PR DESCRIPTION
**Pull request by Lezek123**
- Fix `util:cleanup` script (call `loadDataObjectIdCache` first)
- Allow changing default log level using environment variable (`COLOSSUS_DEFAULT_LOG_LEVEL`)
- Allow adjusting cleanup constants via env (`CLEANUP_MIN_REPLICATION_THRESHOLD`, `CLEANUP_NEW_OBJECT_EXPIRATION_PERIOD`)
- Error handling: Clearer warning messages if unexpected response encountered during sync (ie. 404)
- Add tests to verify cleanup behavior with `--cleanup` flag enabled

Related issue: https://github.com/Joystream/joystream/issues/5182

### CODE REVIEW

File: tests/network-tests/src/utils.ts [hunk 1]
```diff
@@ -143,7 +143,14 @@ export class Utils {
     return new Promise((resolve, reject) => {
       const timeout = setTimeout(() => reject(new Error(`Awaiting ${name} - timeout reached`)), timeoutMs)
       const check = async () => {
-        if (await conditionFunc({ debug })) {
+        let result = false
+        try {
+          result = await conditionFunc({ debug })
+        } catch (e: any) {
+          debug(`conditionFunc error: ${e.toString()}`)
+          return
```
Comment (mnaamani) [inline tests/network-tests/src/utils.ts]: When we reject the promise on timeout the interval is not cleared.

As a general improvement to this `until` function, can we avoid using `setInterval`.
I prefer this approach: 

```ts
const check = async() => {
    // ... do checks
    debug('Condition not satisfied, waiting...')
    setTimeout(check, intervalMs)
}
check()
```

The benefit is one less potentially dangling interval to deal with,  calls to check() usually come more predictably at fixed intervals. We also get rid of potential of using `interval` before it is assigned.

File: storage-node/src/services/sync/tasks.ts [hunk 2]
```diff
@@ -138,9 +138,9 @@ export class DownloadFileTask implements SyncTask {
         .set('X-COLOSSUS-HOST-ID', this.hostId) as unknown as NodeJS.ReadableStream
       const fileStream = fs.createWriteStream(tempFilePath)
 
-      request.on('response', (res) => {
-        if (!res.ok && res.statusCode !== 404) {
-          logger.warn(`Sync - unexpected status code(${res.statusCode}) for ${res?.request?.url}`)
+      request.on('response', (res: Response) => {
+        if (!res.ok) {
+          request.emit('error', `request failed: ${res.error}`)
```
Comment (mnaamani) [inline storage-node/src/services/sync/tasks.ts:143]: Seeing that request is a `Stream` therefore I assume emitting error here will throw an exception in later part of the code.. at `await streamPipeline(request, fileStream)` ?

**Review by mnaamani — COMMENTED** [review]
The new test looks good and the fix for the cleanup command looks correct.
Just left some minor point on the `until` helper function that could be fixed.
Otherwise this is good to go,

File: storage-node/src/services/sync/tasks.ts (hunk 2 above)
Comment (Lezek123) [reply storage-node/src/services/sync/tasks.ts:143]: That's correct, I also tested this specific case locally.

**Review by mnaamani — APPROVED** [review]

**Pull request merged**